test/libyul/yulOptimizerTests/unusedFunctionReturnParameterPruner/pop.yul

```diff
@@ -0,0 +1,27 @@
+// A test to see if the optimization step can deal with pop
+// Expression splitter converts `pop(f(a))` into
```

Comment:
This doesn't work yet. Need to figure out a fix.

Reply:
The UnusedPruner should remove the `pop(z)`

Reply:
UnusedPruner is actually converting it back to `pop(f())`

```
{
	let a := f()
	pop(a)
	function f() -> y {
		sstore(1, sload(1))
	}
}
// ----
// step: unusedPruner
//
// {
//     pop(f())
//     function f() -> y
//     { sstore(1, sload(1)) }
// }
```